torchrl/objectives/value/functional.py

```diff
@@ -38,22 +43,60 @@ def _transpose_time(fun):
     If not -2, makes a transpose of all the multi-dim input tensors to bring
     time at -2, and does the opposite transform for the outputs.
     """
+    ERROR = (
+        "The tensor shape and the time dimension are not compatible: "
+        "got {} and time_dim={}."
+    )
 
     @wraps(fun)
     def transposed_fun(*args, time_dim=-2, **kwargs):
         def transpose_tensor(tensor):
-            if isinstance(tensor, (torch.Tensor, MemmapTensor)) and tensor.ndim >= 2:
-                tensor = tensor.transpose(time_dim, -2)
-            return tensor
+            if (
+                not isinstance(tensor, (torch.Tensor, MemmapTensor))
+                or tensor.numel() <= 1
```

Comment:
Handling `tensor.numel() <= 1` here differently introduces surprising and maybe unwanted behavior. `torch.as_tensr([1])` will not be transformed and has shape [1] (one-dimensional), whereas torch.as_tensor([1, 2])` will be transformed to shape [2, 1] (two-dimensional).

Reply:
Sequences of a single element are defo an edge case. 
A way out could be to assert that sequences have more than one element in each method, when we check the shapes. 
But giving a `gamma=torch.tensor([1.0])` must be supported